konrad/radiation/rrtmg.py

```diff
@@ -262,6 +268,44 @@ def radiative_fluxes(self, atmosphere, surface, cloud):
 
         return lw_fluxes, sw_fluxes
 
+    def calc_cloudy_nomcica_radiation(self, atmosphere, surface, cloud):
+
+        cloud_fraction = deepcopy(cloud.cloud_area_fraction_in_atmosphere_layer)
+
+        if self._state_sw is None:  # first time only
+            cf_cloudy = cloud_fraction[cloud_fraction != 0]
+            if cf_cloudy.shape[0] != 0:
+                if not np.all(cf_cloudy == cf_cloudy[0]):
+                    logging.warning(
+                        'The konrad implementation of nomcica with partial '
+                        'cloud fraction is only suitable for a single '
+                        'rectangular cloud. Consider using mcica instead.')
```

Comment:
Is it really necessary that the cloud is rectangular or is it sufficient if it is a single cloud?

Reply:
Actually the important point is that wherever there is cloud, the cloud fraction is the same. So, there could be two rectangular clouds with the same cloud fraction with some clear sky in between, but with the clouds on top of each other (overlap = maximum) and that would also work. The method only uses a single non-zero cloud fraction, so if the cloud fraction profile input is something else, then the cloud shape is ignored and only a single cloud fraction is used for the cloudy layers.